kv/kvclient/kvcoord: TestDistSenderReplicaStall failed

kv/kvclient/kvcoord.TestDistSenderReplicaStall [failed](https://tanzanite.cluster.engflow.com/invocations/default/d3ccecba-14a9-4b61-aca2-d207b1d71b94?testReportRun=20&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Zjb29yZDprdmNvb3JkX3Rlc3Q=) on master @ [ecfb74b7ad75327f11814966a2cdab1b9793a549](https://github.com/cockroachdb/cockroach/commits/ecfb74b7ad75327f11814966a2cdab1b9793a549):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.RunValues[...].func1
        	            	  | 	pkg/testutils/subtest.go:23
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) aborted in DistSender
        	            	Wraps: (3) result is ambiguous: context deadline exceeded
        	            	Wraps: (4) secondary error attachment
        	            	  | aborted during DistSender.Send: context deadline exceeded
        	            	  | (1)
        	            	  |   | (opaque error wrapper)
        	            	  |   | type name: github.com/cockroachdb/errors/withstack/*withstack.withStack
        	            	  |   | reportable 0:
        	            	  |   |
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendToReplicas
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:3124
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).sendPartialBatch
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:2211
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).divideAndSendBatchToRanges
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1746
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*DistSender).Send
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender.go:1265
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*CrossRangeTxnWrapperSender).Send
        	            	  |   | 	pkg/kv/db.go:224
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).sendUsingSender
        	            	  |   | 	pkg/kv/db.go:1149
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).send
        	            	  |   | 	pkg/kv/db.go:1132
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.sendAndFill
        	            	  |   | 	pkg/kv/db.go:941
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Run
        	            	  |   | 	pkg/kv/db.go:964
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Get
        	            	  |   | 	pkg/kv/db.go:349
        	            	  |   | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord_test.TestDistSenderReplicaStall.func1
        	            	  |   | 	pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go:115
        	            	  |   | github.com/cockroachdb/cockroach/pkg/testutils.RunValues[...].func1
        	            	  |   | 	pkg/testutils/subtest.go:23
        	            	  |   | testing.tRunner
        	            	  |   | 	GOROOT/src/testing/testing.go:1689
        	            	  |   | runtime.goexit
        	            	  |   | 	src/runtime/asm_amd64.s:1695
        	            	  | Wraps: (2) aborted during DistSender.Send
        	            	  | Wraps: (3) context deadline exceeded
        	            	  | Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) context.deadlineExceededError
        	            	Wraps: (5) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) *kvpb.AmbiguousResultError (4) *secondary.withSecondaryError (5) context.deadlineExceededError
        	Test:       	TestDistSenderReplicaStall/clientTimeout=false
    --- FAIL: TestDistSenderReplicaStall/clientTimeout=false (30.11s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=20</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderReplicaStall.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47533

--- a/pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_circuit_breaker_test.go
@@ -22,6 +22,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -36,7 +37,16 @@ import (
 // replica and redirect read requests when the lease moves elsewhere.
 func TestDistSenderReplicaStall(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	defer log.Scope(t).Close(t)
+	scope := log.Scope(t)
+	defer scope.Close(t)
+
+	// See https://github.com/cockroachdb/cockroach/issues/140957.
+	// If this test fails again, we should investigate the issue further, as
+	// it could indicate a failure of the DistSender to circuit break.
+	{
+		skip.UnderDuress(t)
+		defer testutils.StartExecTrace(t, scope.GetDirectory()).Finish(t)
+	}
 
 	testutils.RunTrueAndFalse(t, "clientTimeout", func(t *testing.T, clientTimeout bool) {
 		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)

